### PR TITLE
Implement C2C tests

### LINKED ISSUE #487: C2C feature testing and implementation
## Problem

We want to test C2C inner transaction call described in [c2c-feature-branch](https://github.com/algorand/go-algorand/tree/feature/contract-to-contract/data/transactions/logic#inner-transactions), and the discussion/design-doc is [here](https://docs.google.com/document/d/1uGM62VsN1F4aSTNg4t2q21JmuwS-OO-7bFY0VcaZcro/edit#).

We might want to have some deep-calls (a chain of inner-trans call) as testing method, which should be designed in SDK testing. The ideal result is that we can walk through inner transactions cascaded in tree-like structure.

### PR DESCRIPTION
**Pull request by algochoi**
- [x]  Add pendingTransactionInformation field to ABI methodResult
- [x]  Implement all c2c.feature tests
- [x]  Implement new test in applications.verified
- [x]  Implement new test in compile.feature
- [x]  Deprecate template tests - related PR [here](https://github.com/algorand/js-algorand-sdk/pull/428/files) and related issue: #494 
- [x]  Revert branch change in `run_docker.sh`

Closes #487 #494

### CODE REVIEW

File: tests/cucumber/browser/test.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
 const assert = require('assert');
-const sha256 = require('js-sha256');
```
Comment (tzaffi) [inline tests/cucumber/browser/test.js:2]: was this a bug in the test implementation that you are fixing?

**Review by tzaffi — APPROVED** [review]
Left a non-blocking question. 

LGTM

File: tests/cucumber/browser/test.js (hunk 1 above)
Comment (algochoi) [reply tests/cucumber/browser/test.js:2]: It was an import used by the template tests so I just removed it in this PR

File: src/composer.ts [hunk 2]
```diff
@@ -658,6 +659,7 @@ export class AtomicTransactionComposer {
             methodResult.rawReturnValue
           );
         }
+        methodResult.txInfo = pendingInfo;
```
Comment (jasonpaulos) [inline src/composer.ts:662]: Can you put this line immediately after `pendingInfo` is created? The reason being an error might be thrown between its current placement and the assignment of `pendingInfo`, meaning the caller would not receive `txInfo` when they should.

File: src/main.ts [hunk 3]
```diff
@@ -2,7 +2,6 @@ import * as nacl from './nacl/naclWrappers';
 import * as address from './encoding/address';
 import * as encoding from './encoding/encoding';
 import * as txnBuilder from './transaction';
-import * as LogicTemplatesCommonJSExport from './logicTemplates';
```
Comment (jasonpaulos) [inline src/main.ts:5]: Unfortunately we can't remove the template exports without it being a breaking change -- this is akin to deleting the templates.

**Review by jasonpaulos — CHANGES_REQUESTED** [review]
Only a few small but important comments

**Review by jasonpaulos — APPROVED** [review]

**Pull request merged**